### DISCUSSION TITLE
How to control transparency of `unlit` shading model?

### DISCUSSION THREAD

**Discussion by ndming** [google/filament#6371] [Q&A] [answered]
I'm playing around with 2D geometries on Android and I want to fade out a triangle. The scene only has a 2D triangle and the skybox with hardcoded color values. I use the following material:
```
material {
    name : dynamic_color,

    parameters : [
        {
            type : float4,
            name : baseColor
        },
    ],

    shadingModel : unlit,

    blending : transparent,
}

fragment {
    void material(inout MaterialInputs material) {
        prepareMaterial(material);

        material.baseColor = materialParams.baseColor;
    }
}
```

In the code, I attempt to control the alpha component of `baseColor`:
```
instance.setParameter("baseColor", Colors.RgbaType.SRGB, color.red, color.green, color.blue, 0.5f)
```

However, the triangle is not transparent at all. How can I control transparency in `unlit` shading model? Do I need to switch to `lit` model instead?

I am using Filament v1.27.1

**Comment by romainguy** [comment, marked as answer]
Transparency works with unlit. However, as specified in the [docs](https://google.github.io/filament/Materials.html), you need to return colors with premultiplied (or associated) alpha. So add material.baseColor.rgb *= material.baseColor.a at the end.

**Reply by ndming** [reply]
Turn out the problem has nothing to do with Filament. 
In the code, I actually supply the alpha component as a variable and due to a silly bug, it gets reset to 1.0f by the time the `setParameter` gets invoked.
The material is working fine and I also don't need to add that extra multiplication line at the end.
Thanks for the answer anyway.

**Reply by romainguy** [reply]
Filament exoects that multiplication for blending to work correctly.

**Reply by ndming** [reply]
So I will need to add that multiplication in the fragment code right? I just thought that if I specify `Colors.RgbaType.SRGB` in the `setParameter` function then Filament will handle converting the supplied color to linear SRGB and the multiplication also.

**Reply by romainguy** [reply]
You can do the multiplication on the CPU like this yes. It just needs to happen somewhere.

**Reply by ndming** [reply]
Thank you, I'll keep this in mind.
